pkg/util/hardware/hw_utils.go

```diff
@@ -59,6 +59,9 @@ func ParseCPUSetLine(cpusetLine string) (cpusList []int, err error) {
 			}
 			cpusList = append(cpusList, cpuNum)
 		}
+		if len(cpusList) > limit {
+			return nil, fmt.Errorf("rejecting expanding CPU array for safety reasons, limit is %v", limit)
```

Comment:
Wouldnt it be better to do the safety check before allocating the array? I.e. inside `if len(cpuRange) > 1 {...}`:
```go
if len(cpuRange) > 1 {
			start, err := strconv.Atoi(cpuRange[0])
			if err != nil {
				return nil, err
			}
			end, err := strconv.Atoi(cpuRange[1])
			if err != nil {
				return nil, err
			}
                        // do not allow to exceed the limit
                        if len(cpusList) > limit {
			        return nil, fmt.Errorf("rejecting expanding CPU array for safety reasons, limit is %v", limit)
                        }                        
                        ...
}
```

Reply:
yes, I missed the embedded loop. :+1:

Reply:
Fixed. Also changed the unit test in a way that it would bring down the machine or at least loop for ages.